packages/liveblocks-core/src/comments/comment-body.ts

```diff
@@ -632,3 +632,112 @@ export async function stringifyCommentBody(
 
   return blocks.join(separator);
 }
+
+export type TransformCommentBodyOptions<U extends BaseUserMeta = DU> = {
+  /**
+   * Which format to transform the comment body to.
+   */
+  format?: "html" | "json";
+  /**
+   * A function that returns info from user IDs.
+   */
+  resolveUsers?: (
+    args: ResolveUsersArgs
+  ) => OptionalPromise<(U["info"] | undefined)[] | undefined>;
+};
+
+export type CommentBodyJsonMention = {
+  type: "mention";
+  user: string;
+};
+export type CommentBodyJsonLink = CommentBodyLink;
+export type CommentBodyJsonText = CommentBodyText;
+
+export type CommentBodyJsonInlineElement =
+  | CommentBodyJsonMention
+  | CommentBodyJsonLink
+  | CommentBodyJsonText;
+export type CommentBodyJsonParagraph = {
+  type: "paragraph";
+  children: CommentBodyJsonInlineElement[];
+};
+export type CommentBodyJsonBlockElement = CommentBodyJsonParagraph;
+/**
+ * A `CommentBodyJson` represents the content of a `CommentBody`
+ * with resolved users or not and with absolute urls
+ */
+export type CommentBodyJson = {
+  version: 1;
+  content: CommentBodyJsonBlockElement[];
+};
+
+/**
+ * Transform a `CommentBody` into either an HTML format (HTML safe string)
+ * or into a JSON format and resolves users.
+ *
+ * By default the format is set to "HTML".
+ */
+export async function transformCommentBody(
+  body: CommentBody,
+  options?: TransformCommentBodyOptions<BaseUserMeta>
+): Promise<string | CommentBodyJson> {
+  const format = options?.format ?? "html";
+
+  switch (format) {
+    case "html":
+      return await stringifyCommentBody(body, {
+        format: "html",
+        resolveUsers: options?.resolveUsers,
+      });
```

Comment:
I think I understand what you mean. 
Yes we can keep separate and make the decision to call one or the other in `getThreadNotificationUnreadComments`. 

Then I would rename this function to something like `jsonifyCommentBody` or so but I do not think we should call it with `resolve`. Because following the code intent of `stringifyCommentBody` we should the resolver option optional to stay aligned we what we had already offered to developers. 

WDYT?